packages/workspace/src/node/default-workspace-server.ts

```diff
@@ -162,8 +166,13 @@ export class DefaultWorkspaceServer implements WorkspaceServer {
         }
     }
 
-    protected getUserStoragePath(): string {
-        return path.resolve(os.homedir(), '.theia', 'recentworkspace.json');
+    protected async getUserStoragePath(): Promise<string> {
+        return new Promise<string>(resolve => {
+            this.envServer.getConfigDirUri()
+                .then(configDirUri => new URI(configDirUri))
+                .then(uri => uri.resolve('recentworkspace.json'))
```

Comment:
here promises are used for not async code

Reply:
It's a manner of taste, isn't it? What is the problem with it, please share, so I can catch up? Thanks!

Reply:
it is not executed immediately, but reads to miniqueue of event loop each time? also it is more readable and does not confuse with promises:
```ts
const configDirUir = await this.envServer.getConfigDirUri();
return new URI(configDirUri).resolve('recentworkspace.json');
```